cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/wizard-basic/management-credential-step/ManagementCredentialOneTime.test.tsx

```diff
@@ -0,0 +1,158 @@
+// React imports
+import React from 'react';
+
+// Library imports
+import { fireEvent, render, screen } from '@testing-library/react';
+import ManagementCredentialOneTime from './ManagementCredentialOneTime';
+
+// App imports
+
+const regionsMock = [
+    'us-east-1',
+    'us-east-2',
+    'us-west-1',
+    'us-west-2',
+    'eu-central-1',
+    'eu-east-1',
+    'eu-east-2',
+    'ap-east-1',
+    'ap-south-1',
+    'ca-central-1',
+];
+describe('ManagementCredentialOneTime', () => {
+    const useFormMock = {
+        ...jest.requireActual('react-hook-form'),
+        useForm: () => ({
+            formState: {
+                errors: {},
+            },
+            setValue: jest.fn(),
+            register: jest.fn(),
+        }),
+    };
+    it('should render', async () => {
+        const view = render(
+            <ManagementCredentialOneTime
+                initialRegion={'us-east-1'}
+                handleInputChange={jest.fn()}
+                handleSelectRegion={jest.fn}
+                regions={regionsMock}
+                methods={useFormMock.useForm()}
+            />
+        );
+        expect(view).toBeDefined();
+    });
+    it('initial values', () => {
+        render(
+            <ManagementCredentialOneTime
+                initialSecretAccessKey={'mockSecretAccessKey'}
+                initialSessionToken={'mockSessionToken'}
+                initialAccessKeyId={'mockAccessId'}
+                initialRegion={'us-east-1'}
+                handleInputChange={jest.fn()}
+                handleSelectRegion={jest.fn}
+                regions={regionsMock}
+                methods={useFormMock.useForm()}
+            />
+        );
+        const values = ['mockSecretAccessKey', 'mockSessionToken', 'mockAccessId'];
```

Comment:
nit: might be nice to define these as constants first, and then use them both in the ManagementCredentialOneTime as well as here. Also, I would name them 'testSecretAccessKey' or some such. I associate 'mock' with a function that actually mocks out a real operation.
This is very nitpicky, but just mentioning because I think small naming things like this can help make the test more readable, especially if everyone follows the same naming conventions

Reply:
@swalner-vmware the values variable contains all test data for the input fields in the component. I'm curious that why we want to use it in the ManagementCredentialOneTime? As default values for each input field?

Reply:
fixed.